lib/logger.js

```diff
@@ -70,50 +70,48 @@ function now () {
 }
 
 function createLogger (options) {
-  if (isValidLogger(options.logger)) {
+  if (!options.logger) {
+    const logger = nullLogger
+    logger.child = () => logger
+    return { logger, hasLogger: false }
+  }
+
+  if (implementsLog4jInterface(options.logger)) {
+    if (!options.logger.child) throw FST_ERR_LOG_INVALID_OBJECT('child')
+
     const logger = createPinoLogger({
       logger: options.logger,
       serializers: Object.assign({}, serializers, options.logger.serializers)
     })
     return { logger, hasLogger: true }
-  } else if (!options.logger) {
-    const logger = nullLogger
-    logger.child = () => logger
-    return { logger, hasLogger: false }
-  } else {
-    const localLoggerOptions = {}
-    if (Object.prototype.toString.call(options.logger) === '[object Object]') {
-      Reflect.ownKeys(options.logger).forEach(prop => {
-        Object.defineProperty(localLoggerOptions, prop, {
-          value: options.logger[prop],
-          writable: true,
-          enumerable: true,
-          configurable: true
-        })
-      })
-    }
-    localLoggerOptions.level = localLoggerOptions.level || 'info'
-    localLoggerOptions.serializers = Object.assign({}, serializers, localLoggerOptions.serializers)
-    options.logger = localLoggerOptions
-    const logger = createPinoLogger(options.logger)
-    return { logger, hasLogger: true }
   }
-}
 
-function isValidLogger (logger) {
-  if (!logger) {
-    return false
+  const localLoggerOptions = {}
+  if (Object.prototype.toString.call(options.logger) === '[object Object]') {
+    Reflect.ownKeys(options.logger).forEach(prop => {
+      Object.defineProperty(localLoggerOptions, prop, {
+        value: options.logger[prop],
+        writable: true,
+        enumerable: true,
+        configurable: true
+      })
+    })
   }
+  localLoggerOptions.level = localLoggerOptions.level || 'info'
+  localLoggerOptions.serializers = Object.assign({}, serializers, localLoggerOptions.serializers)
+  options.logger = localLoggerOptions
+  const logger = createPinoLogger(options.logger)
+  return { logger, hasLogger: true }
+}
 
-  let result = true
-  const methods = ['info', 'error', 'debug', 'fatal', 'warn', 'trace', 'child']
+function implementsLog4jInterface (logger) {
```

Comment:
if the logged does not have the `trace` method, I think the issue is still open

Instead of `return true`, we should count how many methods matches and return the methods that are missing

In this way the error will tell us precisely what is missing

Reply:
I would still prefer isValidLogger over implementsLog4jInterface. Also instead of returning false we could directly throw which method is missing. E.g. 
```js
throw new TypeError(`Invalid logger instance. The logger instance should have a '${method}' method.`)
```